### PR TITLE
feat(Python): Format docstrings

### PR DESCRIPTION
**Pull request by lucasmcdonald3**
*Issue #, if available:*

*Description of changes:*

Format docstrings.

See https://aws-cryptographic-material-providers-library.readthedocs.io/en/latest/generated/aws_cryptographic_materialproviders.smithygenerated.aws_cryptography_keystore.models.html for an example.
Before, the parameters and function descriptions were not displaying correctly because the docstrings weren't formatted.
Now, they are appearing correctly.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithypython/localservice/extensions/DafnyPythonLocalServiceStructureGenerator.java [hunk 1]
```diff
@@ -374,13 +374,6 @@ protected void writeFromDict(boolean isError) {
       () -> {
         writer.writeDocs(() -> {
           writer.write("Creates a $L from a dictionary.\n", shapeName);
-          writer.write(
```
Comment (robin-aws) [inline codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithypython/localservice/extensions/DafnyPythonLocalServiceStructureGenerator.java:377]: This is unrelated removal of comments that weren't supposed to be user-facing in the first place?

**Review by robin-aws — APPROVED** [review]

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithypython/localservice/extensions/DafnyPythonLocalServiceStructureGenerator.java (hunk 1 above)
Comment (lucasmcdonald3) [reply codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithypython/localservice/extensions/DafnyPythonLocalServiceStructureGenerator.java:377]: This comment is inaccurate for Python codegen (and is inaccurate for Smithy-Python codegen).
* Formatting according to docstring note: `SomeMemberName`
  * This is the "boto3" formatting.
* Actual Smithy-Python formatting: `someMemberName`
* Smithy-Dafny Python formatting: `some_member_name`
